sdk/python/kfp/containers/_build_image_api.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the speci
+
+__all__ = [
+    'build_image_from_working_dir',
+    'default_base_image',
+    'get_python_image_for_current_version',
+]
+
+
+import logging
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+import requests
+
+from . import get_default_image_builder
+from ..compiler._container_builder import ContainerBuilder
+
+
+def get_python_image_for_current_version() -> str:
+    return 'python' + ':{}.{}.{}'.format(sys.version_info[0], sys.version_info[1], sys.version_info[2])
+
+
+#default_base_image = get_python_image_for_current_version()
+default_base_image = 'gcr.io/deeplearning-platform-release/tf-cpu.1-14' # The size of this image is 4.35GB which really concerns me. The GPU image size is 6.45GB.
+
+
+_container_work_dir = '/python_env'
+
+
+def generate_dockerfile_text(context_dir: str, dockerfile_path: str, base_image: str = None):
+    # Generating the Dockerfile
+    logging.info('Generating the Dockerfile')
+
+    requirements_rel_path = 'requirements.txt'
+    requirements_path = os.path.join(context_dir, requirements_rel_path)
+    requirements_file_exists = os.path.exists(requirements_path)
+
+    if not base_image:
+        base_image = default_base_image
+    if callable(base_image):
+        base_image = base_image()
+
+    dockerfile_lines = []
+    dockerfile_lines.append('FROM {}'.format(base_image))
+    dockerfile_lines.append('WORKDIR {}'.format(_container_work_dir))
+    if requirements_file_exists:
+        dockerfile_lines.append('COPY {} .'.format(requirements_rel_path))
+        dockerfile_lines.append('RUN python3 -m pip install -r {}'.format(requirements_rel_path))
+    dockerfile_lines.append('COPY . .')
+
+    return '\n'.join(dockerfile_lines)
+
+
+def build_image_from_working_dir(image_name : str = None, working_dir : str = None, file_filter_re : str = '.*\.py',  timeout : int = 1000, base_image: str = None, builder : ContainerBuilder = None) -> str:
```

Comment:
As discussed, making this function returns back a delayed function.

Reply:
What do you think about the following way that we've discussed?
```
kfp.components.default_base_image = lambda: build_image_from_working_dir('./my_dir')
```

Reply:
lgtm. Make sure we use the syntax in our sample and documentation. It's not obvious that user can discover the syntax by themselves.